frontend/src/javascript/utils/intervalsUtils.js

```diff
@@ -1,5 +1,50 @@
-export function areIntervalsIntersect(aStart, aEnd, bStart, bEnd) {
-    return aEnd > bStart && aStart < bEnd;
+export function areIntervalsIntersectOrTouching(aStart, aEnd, bStart, bEnd) {
+    return aEnd >= bStart && aStart <= bEnd;
+}
+
+export function applyChangeOnInterval(aStart, aEnd, change) {
```

Comment:
Trochę mnie kłuje w oczy, że jest tutaj mieszany standard tego, że dla jednego przedziału przekazujemy początek i koniec w dwóch zmiennych, drugi przedział jako obiekt. Change ma też inne pola niż start i end, więc rozdzielanie tego na pojedyncze pola to by było strzałem w kolano, ale aStart i aEnd można połączyć w jeden obiekt. Te 2 pozostałe funkcje w sumie też można by było zmienić, że przekazujemy 2 obiekty i każdy z nich ma pola .start i .end.